init.js

```diff
@@ -88,6 +84,24 @@ require.cache[SHMPath].exports = class WebSocketManager extends SHM {
 
 		const [shard] = this.shardQueue;
 
+		// Pushes shards that require reidentifying to the back of the queue
+		const hotReload = this.client.options.hotReload;
+		if (hotReload) {
+			const data = (hotReload.sessionData || this.client._loadSession(shard.id))?.[shard.id]
+			if(data?.id && data.sequence > 0 && !shard.sessionID && data.shardCount === this.totalShards && data.lastConnected + 60000 > Date.now()) {
+				shard.sessionID = data.id;
+				shard.closeSequence = shard.sequence = data.sequence;
+				this.debug("Loaded sessions from cache, resuming previous session.", shard);
+			}
+			else if (this.shardQueue.size > 1 && !shard.requeued) {
+				shard.requeued = true;
+				this.shardQueue.delete(shard);
+				this.shardQueue.add(shard);
+				this.debug("Shard required to identify, pushed to the back of the queue.", shard);
+				return this.createShards();
+			}
+		}	
+
 		this.shardQueue.delete(shard);
```

Comment:
isnt this deleting requeued shards?

Reply:
No, because it returns `this.createShards()` it will never get to deleting it. If the shard has sessions then it will delete it from the queue like normal but if it requires a reidentify it removed it from the start of the queue and adds it to the end and then creates that shard again.

Reply:
am blind